plugins/java/connectors/imap-connector/src/main/java/com/lucidworks/fusion/connector/plugin/client/ImapClient.java

```diff
@@ -0,0 +1,167 @@
+package com.lucidworks.fusion.connector.plugin.client;
+
+import com.lucidworks.fusion.connector.plugin.ImapConfig;
+import org.apache.logging.log4j.Logger;
+
+import javax.inject.Inject;
+import javax.mail.*;
+import javax.mail.search.FlagTerm;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.UUID;
+
+public class ImapClient {
+  private Store store;
+
+  @Inject
+  public ImapClient(ImapConfig config, Logger logger) throws MessagingException {
+    // extract properties
+    String host = config.getProperties().getHost();
+    String username = config.getProperties().getUsername();
+    String password = config.getProperties().getPassword();
+    boolean ssl = config.getProperties().getSsl();
+
+    String provider;
+    if (ssl) {
+      provider = "imaps";
+    } else {
+      provider = "imap";
+    }
+
+    Properties props = new Properties();
+    props.setProperty("mail.store.protocol", provider);
+
+    // connect to imap server
+    Session session = Session.getDefaultInstance(props, null);
+    store = session.getStore(provider);
+    store.connect(host, username, password);
+  }
+
+  public List<Email> getMessages(String folderName) throws MessagingException, IOException {
+    validConnection();
+
+    Folder folder = store.getFolder(folderName);
+    folder.open(Folder.READ_ONLY);
+    Message[] messages = folder.getMessages();
+    List<Email> emails = new ArrayList<>();
+    for(Message message : messages) {
+      emails.add(toEmail(message));
```

Comment:
Please use lambda here to iterate the messages, and the method `.map` to convert the `Message` to `Email`, then use `.collect(Collectors.toList())` to set the emails